tensilelite/Tensile/Common/RegisterPool.py

```diff
@@ -236,8 +237,11 @@ def checkOutAligned(self, size, alignment, tag="_untagged_aligned_", preventOver
       oldSize = len(self.pool)
       if self.occupancyLimitSize > 0:
         if newSize > self.occupancyLimitSize and newSize <= self.occupancyLimitMaxSize:
-          print("newSize", newSize, "OldSIze", oldSize, "Limit", self.occupancyLimitSize)
-          assert self.occupancyLimitSize >= newSize
+          print(self.type, "newSize", newSize, "OldSIze", oldSize, "Limit", self.occupancyLimitSize)
+          if self.states == None:
+            assert self.occupancyLimitSize >= newSize
+          else:
+            self.states.overflowedResources = 4
```

Comment:
Don't bypass the guard, this is meant to fail.

Reply:
No, this kernel will be rejected.
Writer will check the overflowedResources and reject this kernel.